.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,49 @@
+name: Deploy instance
+on:
+    push:
+        branches:
+            - main
+jobs:
+    deploy:
+        runs-on: ubuntu-latest
+        timeout-minutes: 15
+        env:
+            SECRET_MASTER_KEY: ${{ secrets.SECRET_MASTER_KEY }}
+            SECRET_JWT_SECRET: ${{ secrets.SECRET_JWT_SECRET }}
+            DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+            DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+        steps:
+            - name: Checkout Code
+              uses: actions/checkout@v4
+
+            - name: Setup deploy file
+              run: echo ${{ secrets.KAMAL_DEPLOY_YAML }} | base64 -d > config/deploy.yml
```

Comment:
For those wondering what & why, the deploy.yml file with the server IP is base 64 encoded as a secret in Github. This is to hide the IP address from the public until the firewalls are correctly in place for whom is allowed to connect to the IP.